<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
  <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
  <meta http-equiv="Content-Style-Type" content="text/css" />
  <meta name="generator" content="pandoc" />
  <meta name="author" content="David Montaner" />
  <title>NGS: Variant Annotation</title>
  <style type="text/css">code{white-space: pre;}</style>
  <style>
  
  html {
      color:#000;
      background:#FFF;
  }
  
  body,div,dl,dt,dd,ul,ol,li,h1,h2,h3,h4,h5,h6,pre,code,form,fieldset,legend,input,textarea,p,blockquote,th,td {
      margin:0;
      padding:0;
  }
  
  table {
      border-spacing:0;
      font-size:inherit;
      font:100%;
      /*max-width:100%;*/
      background-color:transparent;
      border:1px solid #DDD;
      border-left:0;
      border-collapse:collapsed;
      -webkit-border-radius:4px;
      -moz-border-radius:4px;
      border-radius:4px;
  }
  
  fieldset,img {
      border:0;
  }
  
  address,caption,cite,code,dfn,em,strong,th,var {
      font-style:normal;
      font-weight:400;
  }
  
  ol,ul {
      list-style:none;
  }
  
  caption,th {
      text-align:left;
  }
  
  h1,h2,h3,h4,h5,h6 {
      font-size:100%;
      font-weight:400;
      color:#20639b;
      margin:1em 0;
  }
  
  q:before,q:after {
      content:'';
  }
  
  abbr,acronym {
      border:0;
      font-variant:normal;
      border-bottom:1px dotted #000;
      cursor:help;
  }
  
  sup {
      vertical-align:text-top;
  }
  
  sub {
      vertical-align:text-bottom;
  }
  
  input,textarea,select {
      font-family:inherit;
      font-weight:inherit;
      font-size:100%;
  }
  
  legend {
      color:#000;
  }
  
  h1,h2,h3,h4,h5,h6,strong {
      font-weight:700;
  }
  
  em {
      font-style:italic;
  }
  
  blockquote,ul,ol,dl {
      margin:1em;
  }
  
  ol,ul,dl {
      margin-left:2em;
  }
  
  ol {
      list-style:decimal outside;
  }
  
  ul {
      list-style:disc outside;
  }
  
  dl dd {
      margin-left:1em;
  }
  
  th,td {
      line-height:1.125em;
      text-align:left;
      vertical-align:top;
      border-top:1px solid #DDD;
      border-left:1px solid #DDD;
      padding:.5em;
  }
  
  th {
      font-weight:700;
      text-align:center;
      color:#20639b;
  }
  
  table thead:first-child tr:first-child th,table tbody:first-child tr:first-child th,table tbody:first-child tr:first-child td {
      border-top:0;
  }
  
  table thead:first-child tr:first-child th:first-child,table tbody:first-child tr:first-child td:first-child {
      -webkit-border-radius:4px 0 0 0;
      -moz-border-radius:4px 0 0;
      border-radius:4px 0 0 0;
  }
  
  table thead:first-child tr:first-child th:last-child,table tbody:first-child tr:first-child td:last-child {
      -webkit-border-radius:0 4px 0 0;
      -moz-border-radius:0 4px 0 0;
      border-radius:0 4px 0 0;
  }
  
  table thead:last-child tr:last-child th:first-child,table tbody:last-child tr:last-child td:first-child {
      -webkit-border-radius:0 0 0 4px;
      -moz-border-radius:0 0 0 4px;
      border-radius:0 0 0 4px;
  }
  
  table thead:last-child tr:last-child th:last-child,table tbody:last-child tr:last-child td:last-child {
      -webkit-border-radius:0 0 4px 0;
      -moz-border-radius:0 0 4px;
      border-radius:0 0 4px 0;
  }
  
  table tbody tr:nth-child(odd) td,table tbody tr:nth-child(odd) th {
      background-color:#f9f9f9;
  }
  
  caption {
      margin-bottom:.5em;
      text-align:center;
  }
  
  p,fieldset,table,pre {
      margin-top:.5em;
      margin-bottom:.5em;
  }
  
  body {
      font:16px/1.4 "Helvetica Neue", Helvetica, "Open Sans", Arial, sans-serif;
      color:#333;
      /*max-width:960px;*/
      /*margin:0 auto;*/
      margin:1em;
  }
  
  pre,code,kbd,samp,tt {
      font-family:monospace;
      font-size:90%;
      line-height:100%;
  }
  
  h1 {
      font-size:200%;
      border-bottom:2px dotted #ddd;
  }
  
  a,a:visited,code {
      /*color:#f06024;*/
      color:#20639b;
  }
  
  a:link {text-decoration:none;}    /* unvisited link */
  a:visited {text-decoration:none;} /* visited link */
  a:hover {text-decoration:underline;}   /* mouse over link */
  a:active {text-decoration:underline;}  /* selected link */
  
  
  h2 {
      font-size:160%;
  }
  
  h3 {
      font-size:120%;
  }
  
  h4 {
      font-size:110%;
  }
  
  h5 {
      font-size:95%;
  }
  
  h6 {
      font-size:90%;
      color:#999;
  }
  
  dl {
      margin:0 0 1.125em;
  }
  
  dt {
      font-weight:700;
      line-height:1.125em;
  }
  
  dd {
      margin-left:1em;
      line-height:1.125em;
  }
  
  pre,blockquote {
      border:1px solid #ddd;
      background-color:#f1f1f1;
      -webkit-border-radius:6px;
      -moz-border-radius:6px;
      border-radius:6px;
      padding:10px;
  }
  
  blockquote {
      margin:1em 2.5em;
  }
  
  code {
      color:#20639b;
      background:#f1f1f1;
  }
  
  img {
      max-width:100%;
  }
  
  ol[type=1] {
      list-style-type:decimal;
  }
  
  ol[type=a] {
      list-style-type:lower-alpha;
  }
  
  ol[type=A] {
      list-style-type:upper-alpha;
  }
  
  ol[type=I] {
      list-style-type:upper-roman;
  }
  
  ol[type=i] {
      list-style-type:lower-roman;
  }
  
  
  
  h1.title {
          font-size: 200%;
          font-weight: bold;
          padding-top: 0.2em;
          /*padding-bottom: 0.2em;*/
          text-align: center;
          border: none;
  }
  
  h2.author {
          font-size: 100%;
          text-align: center;
          border: none;
  }
  
  h3.date {
          font-size: 75%;
          text-align: center;
          border: none;
          font-style:italic;
          font-weight: bold;
  }
  
  </style>
</head>
<body>
<div id="header">
<h1 class="title">NGS: Variant Annotation</h1>
<h2 class="author"><a href="http://www.dmontaner.com/">David Montaner</a></h2>
<h3 class="date">(03-12-2013)</h3>
</div>
<!--
Clean the starting dataset
================================================================================
#    rm -r data/*
#    cp data0/f* data/
#    cd data
-->

<h1 id="introduction">Introduction</h1>
<p>Once variants have been called in our dataset, data from many different sources may be pulled to filter the most important of those variants according to our experiment.</p>
<p>See [Ensembl Variation data] (http://www.ensembl.org/info/genome/variation/predicted_data.html)</p>
<p>Two of the most used filters for our variants are SIFT and PolyPhen.</p>
<h2 id="sift">SIFT</h2>
<p>Sorting Intolerant from Tolerant (SIFT) predicts whether an amino acid substitution affects protein function and thus is likely to have a phenotypic effect.</p>
<p>SIFT prediction is based on the degree of conservation of amino acid residues in sequence alignments derived from closely related sequences,</p>
<p>See <a href="http://sift.jcvi.org/">SIFT site</a> for details.</p>
<!-- References -->
<h2 id="polyphen">PolyPhen</h2>
<p>Polymorphism Phenotyping (PolyPhen-2) is a software tool which predicts possible impact of amino acid substitutions on the structure and function of human proteins using straightforward physical and evolutionary comparative considerations.</p>
<p>PolyPhen makes its predictions using UniProt features, PSIC profiles scores derived from multiple alignment and matches to PDP or PQS structural databases.</p>
<h1 id="genomes-exercise">1000 Genomes Exercise</h1>
<p>Go to the 1000 Genomes FTP site:</p>
<p><a href="ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/phase1/analysis_results/integrated_call_sets">ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/phase1/analysis_results/integrated_call_sets</a></p>
<p>The files in this directory represent the final variant call set associated with the phase1 analysis.</p>
<p>You can use <code>tabix</code> to get the header of the VCF file and see the identifiers fore each of the samples:</p>
<pre><code>tabix -H ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/phase1/analysis_results/integrated_call_sets/ALL.chr20.integrated_phase1_v3.20101123.snps_indels_svs.genotypes.vcf.gz &gt; header.txt</code></pre>
<p>Download the first variants in chromosome 20</p>
<pre><code>tabix -h ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/phase1/analysis_results/integrated_call_sets/ALL.chr20.integrated_phase1_v3.20101123.snps_indels_svs.genotypes.vcf.gz 20:1-100000 &gt; first_chr20.vcf</code></pre>
<p>and extract their IDs suing the Linux command <code>cut</code></p>
<pre><code>cut -f 3 first_chr20.vcf &gt; SNPids</code></pre>
<p>See <a href="http://www.1000genomes.org/category/tabix">here</a> for more examples on the usage of <code>tabix</code> for the 1000 Genomes project.</p>
<h2 id="annotation-example-using-sift">Annotation example using SIFT</h2>
<p>Download the precomputed SIFT scores form the FTP of the <a href="http://sift.jcvi.org/">SIFT web page</a></p>
<p><a href="ftp://ftp.jcvi.org/pub/data/sift/dbSNP_132">ftp://ftp.jcvi.org/pub/data/sift/dbSNP_132</a></p>
<p>and try to include the information into the <strong>first_chr20.vcf</strong> file</p>
<p>The scritp <strong>sift_and_polyphen_in_bioconductor.r</strong> shows how to get the same information from Bioconductor precompiled packages.</p>
<h1 id="unknown-variants-exercise">Unknown Variants Exercise</h1>
<p>In our VCF file <strong>f090_snp_filtered.vcf</strong> created in previous sessions there are not dbSNP identifiers. It is likely that most of the variants are novel.</p>
<p>Use the data form <a href="http://sift.jcvi.org/">SIFT web page</a></p>
<p><a href="ftp://ftp.jcvi.org/pub/data/sift/Human_db_37_ensembl_63/">ftp://ftp.jcvi.org/pub/data/sift/Human_db_37_ensembl_63/</a></p>
<p>to find out which of the variants have a SIFT score and filter the variants in the file according to such information.</p>
<p>The scritp <strong>read_sqlite.r</strong> shows how to extract the information form the SQLite file.</p>
</body>
</html>
