<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
  <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
  <meta http-equiv="Content-Style-Type" content="text/css" />
  <meta name="generator" content="pandoc" />
  <meta name="author" content="David Montaner" />
  <title>NGS: Variant Calling</title>
  <style>
  
  html {
      color:#000;
      background:#FFF;
  }
  
  body,div,dl,dt,dd,ul,ol,li,h1,h2,h3,h4,h5,h6,pre,code,form,fieldset,legend,input,textarea,p,blockquote,th,td {
      margin:0;
      padding:0;
  }
  
  table {
      border-spacing:0;
      font-size:inherit;
      font:100%;
      /*max-width:100%;*/
      background-color:transparent;
      border:1px solid #DDD;
      border-left:0;
      border-collapse:collapsed;
      -webkit-border-radius:4px;
      -moz-border-radius:4px;
      border-radius:4px;
  }
  
  fieldset,img {
      border:0;
  }
  
  address,caption,cite,code,dfn,em,strong,th,var {
      font-style:normal;
      font-weight:400;
  }
  
  ol,ul {
      list-style:none;
  }
  
  caption,th {
      text-align:left;
  }
  
  h1,h2,h3,h4,h5,h6 {
      font-size:100%;
      font-weight:400;
      color:#20639b;
      margin:1em 0;
  }
  
  q:before,q:after {
      content:'';
  }
  
  abbr,acronym {
      border:0;
      font-variant:normal;
      border-bottom:1px dotted #000;
      cursor:help;
  }
  
  sup {
      vertical-align:text-top;
  }
  
  sub {
      vertical-align:text-bottom;
  }
  
  input,textarea,select {
      font-family:inherit;
      font-weight:inherit;
      font-size:100%;
  }
  
  legend {
      color:#000;
  }
  
  h1,h2,h3,h4,h5,h6,strong {
      font-weight:700;
  }
  
  em {
      font-style:italic;
  }
  
  blockquote,ul,ol,dl {
      margin:1em;
  }
  
  ol,ul,dl {
      margin-left:2em;
  }
  
  ol {
      list-style:decimal outside;
  }
  
  ul {
      list-style:disc outside;
  }
  
  dl dd {
      margin-left:1em;
  }
  
  th,td {
      line-height:1.125em;
      text-align:left;
      vertical-align:top;
      border-top:1px solid #DDD;
      border-left:1px solid #DDD;
      padding:.5em;
  }
  
  th {
      font-weight:700;
      text-align:center;
      color:#20639b;
  }
  
  table thead:first-child tr:first-child th,table tbody:first-child tr:first-child th,table tbody:first-child tr:first-child td {
      border-top:0;
  }
  
  table thead:first-child tr:first-child th:first-child,table tbody:first-child tr:first-child td:first-child {
      -webkit-border-radius:4px 0 0 0;
      -moz-border-radius:4px 0 0;
      border-radius:4px 0 0 0;
  }
  
  table thead:first-child tr:first-child th:last-child,table tbody:first-child tr:first-child td:last-child {
      -webkit-border-radius:0 4px 0 0;
      -moz-border-radius:0 4px 0 0;
      border-radius:0 4px 0 0;
  }
  
  table thead:last-child tr:last-child th:first-child,table tbody:last-child tr:last-child td:first-child {
      -webkit-border-radius:0 0 0 4px;
      -moz-border-radius:0 0 0 4px;
      border-radius:0 0 0 4px;
  }
  
  table thead:last-child tr:last-child th:last-child,table tbody:last-child tr:last-child td:last-child {
      -webkit-border-radius:0 0 4px 0;
      -moz-border-radius:0 0 4px;
      border-radius:0 0 4px 0;
  }
  
  table tbody tr:nth-child(odd) td,table tbody tr:nth-child(odd) th {
      background-color:#f9f9f9;
  }
  
  caption {
      margin-bottom:.5em;
      text-align:center;
  }
  
  p,fieldset,table,pre {
      margin-top:.5em;
      margin-bottom:.5em;
  }
  
  body {
      font:16px/1.4 "Helvetica Neue", Helvetica, "Open Sans", Arial, sans-serif;
      color:#333;
      /*max-width:960px;*/
      /*margin:0 auto;*/
      margin:1em;
  }
  
  pre,code,kbd,samp,tt {
      font-family:monospace;
      font-size:90%;
      line-height:100%;
  }
  
  h1 {
      font-size:200%;
      border-bottom:2px dotted #ddd;
  }
  
  a,a:visited,code {
      /*color:#f06024;*/
      color:#20639b;
  }
  
  a:link {text-decoration:none;}    /* unvisited link */
  a:visited {text-decoration:none;} /* visited link */
  a:hover {text-decoration:underline;}   /* mouse over link */
  a:active {text-decoration:underline;}  /* selected link */
  
  
  h2 {
      font-size:160%;
  }
  
  h3 {
      font-size:120%;
  }
  
  h4 {
      font-size:110%;
  }
  
  h5 {
      font-size:95%;
  }
  
  h6 {
      font-size:90%;
      color:#999;
  }
  
  dl {
      margin:0 0 1.125em;
  }
  
  dt {
      font-weight:700;
      line-height:1.125em;
  }
  
  dd {
      margin-left:1em;
      line-height:1.125em;
  }
  
  pre,blockquote {
      border:1px solid #ddd;
      background-color:#f1f1f1;
      -webkit-border-radius:6px;
      -moz-border-radius:6px;
      border-radius:6px;
      padding:10px;
  }
  
  blockquote {
      margin:1em 2.5em;
  }
  
  code {
      color:#20639b;
      background:#f1f1f1;
  }
  
  img {
      max-width:100%;
  }
  
  ol[type=1] {
      list-style-type:decimal;
  }
  
  ol[type=a] {
      list-style-type:lower-alpha;
  }
  
  ol[type=A] {
      list-style-type:upper-alpha;
  }
  
  ol[type=I] {
      list-style-type:upper-roman;
  }
  
  ol[type=i] {
      list-style-type:lower-roman;
  }
  
  
  
  h1.title {
          font-size: 200%;
          font-weight: bold;
          padding-top: 0.2em;
          /*padding-bottom: 0.2em;*/
          text-align: center;
          border: none;
  }
  
  h2.author {
          font-size: 100%;
          text-align: center;
          border: none;
  }
  
  h3.date {
          font-size: 75%;
          text-align: center;
          border: none;
          font-style:italic;
          font-weight: bold;
  }
  
  </style>
</head>
<body>
<div id="header">
<h1 class="title">NGS: Variant Calling</h1>
<h2 class="author"><a href="http://www.dmontaner.com/">David Montaner</a></h2>
<h3 class="date">(19-11-2013)</h3>
</div>
<!--
Clean the starting dataset
================================================================================
    rm -r data/*
    cp data0/f* data/
    cd data
-->



<h1 id="preliminaries">Preliminaries</h1>
<h2 id="software-used-in-this-practical">Software used in this practical</h2>
<ul>
<li><a href="http://bio-bwa.sourceforge.net/" title="Burrows-Wheeler Aligner">BWA</a> (Burrows-Wheeler Aligner): A software package for mapping low-divergent sequences against a large reference genome.</li>
<li><a href="http://picard.sourceforge.net">Picard</a>: Java-based command-line utilities to manipulate SAM files.</li>
<li><a href="http://www.broadinstitute.org/gatk/">GATK</a> (Genome Analysis Toolkit): A package to analyze next-generation re-sequencing data, primary focused on variant discovery and genotyping.</li>
</ul>
<!-- Tool References -->















<h2 id="file-formats">File formats:</h2>
<ul>
<li>FastQ: see <a href="http://en.wikipedia.org/wiki/FASTQ_format">Wikipedia</a> and <a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2847217">NAR 2010</a> specifications.</li>
<li>SAM/BAM (Sequence Alignment Map): see <a href="http://samtools.sourceforge.net/SAM1.pdf">SAMtools</a> and <a href="http://genome.ucsc.edu/goldenPath/help/bam.html">UCSC</a> specifications.</li>
<li>GTF: see <a href="http://www.ensembl.org/info/website/upload/gff.html">Ensembl</a> and <a href="http://genome.ucsc.edu/FAQ/FAQformat.html#format4">UCSC</a> specifications.</li>
<li>VCF Variant Call Format: see <a href="http://www.1000genomes.org/wiki/Analysis/Variant%20Call%20Format/vcf-variant-call-format-version-41">1000 Genomes</a> and <a href="http://en.wikipedia.org/wiki/Variant_Call_Format">Wikipedia</a> specifications.</li>
</ul>
<!-- File Format References -->













<h1 id="data">Data</h1>
<p>Our data analysis will start with a reference genome in fasta format:</p>
<ul>
<li>f000_reference.fa</li>
</ul>
<p>In this case it is the chromosome 20 of the human genome; run:</p>
<pre><code>grep &quot;&gt;&quot; f000_reference.fa</code></pre>
<p>and two fastq files from a <em>paired-end</em> experiment:</p>
<ul>
<li>f010_reads1.fastq</li>
<li>f010_reads2.fastq</li>
</ul>
<h1 id="mapping-using-bwa">Mapping using BWA</h1>
<p>First we have to create the index</p>
<pre><code>bwa index -a bwtsw f000_reference.fa</code></pre>
<p>And then we align each of the files. This step is done separately for each file; it will be the same for paired end or single end schemes.</p>
<pre><code>bwa aln -f f020_reads1.sai f000_reference.fa f010_reads1.fastq
bwa aln -f f020_reads2.sai f000_reference.fa f010_reads2.fastq</code></pre>
<p>Now we combine the two SAM files in a combined one (<em>paired-end</em> protocol):</p>
<pre><code>bwa sampe -f f030_aligned.sam -r &#39;@RG\tID:myID\tSM:mySample&#39; f000_reference.fa f020_reads1.sai f020_reads2.sai f010_reads1.fastq f010_reads2.fastq</code></pre>
<p>Convert file to binary BAM format:</p>
<pre><code>samtools view -bS f030_aligned.sam &gt; f040_unordered_aligned.bam</code></pre>
<p>Order by chromosomic position:</p>
<pre><code>samtools sort f040_unordered_aligned.bam f050_aligned</code></pre>
<p>Index the BAM file:</p>
<pre><code>samtools index f050_aligned.bam</code></pre>
<!--
##AddOrReplaceReadGroups ?? 
-->



<h1 id="variant-calling">Variant Calling</h1>
<h2 id="gatk-preliminaries">GATK preliminaries</h2>
<p>Create a sequence dictionary for the reference using <a href="#software-used-in-this-practical">Picard Tools</a>.</p>
<pre><code>## this is just an example path to picard java programs
picard_path=&quot;/home/dmontaner/bin/picard-tools-1.99/picard-tools-1.99&quot;

java -jar $picard_path/CreateSequenceDictionary.jar REFERENCE=f000_reference.fa OUTPUT=f000_reference.dict</code></pre>
<p>Create an index for the reference using <strong>samtools</strong>:</p>
<pre><code>samtools faidx f000_reference.fa</code></pre>
<p>Run <strong>GATK</strong> CountReads to test that all formats are right:</p>
<pre><code>## this is just an example path
gatk_path=&quot;/home/dmontaner/bin/GenomeAnalysisTK-2.7-4-g6f46d11&quot;

java -jar $gatk_path/GenomeAnalysisTK.jar -T CountReads -R f000_reference.fa -I f050_aligned.bam</code></pre>
<!-- GATK Preprocessing -->

<h2 id="indel-realignment">InDel Realignment</h2>
<p>First we create a target list of intervals which need to be realigned:</p>
<pre><code>java -jar $gatk_path/GenomeAnalysisTK.jar -T RealignerTargetCreator -R f000_reference.fa -I f050_aligned.bam -o f060_intervals2realign_aligned.list</code></pre>
<!--
Revise:
-I dedup_aligned.bam \ 
-L 20 \ 
-known gold_indels.vcf \ 
-->


<p>Then we perform a realignment of the selected intervals:</p>
<pre><code>java -jar $gatk_path/GenomeAnalysisTK.jar -T IndelRealigner -R f000_reference.fa -I f050_aligned.bam -targetIntervals f060_intervals2realign_aligned.list -o f070_realigned_aligned.bam</code></pre>
<!--
Base Recalibration
------------------

parece que el bam esta malformado::: faltara lo de recomprimirlo

  GenomeAnalysisTK.jar -T BaseRecalibrator -R f000_reference.fa -I f070_realigned_aligned.bam -knownSites 0_dbsnp_chr2.vcf -o f080_recalibration.table

  GenomeAnalysisTK.jar -T BaseRecalibrator -R f000_reference.fa -I f070_realigned_aligned.bam -knownSites mi.vcf -o f080_recalibration.table

  GenomeAnalysisTK.jar -T BaseRecalibrator -R f000_reference.fa -I f070_realigned_aligned.bam -knownSites mi.vcf -o f080_recalibration.table

  GenomeAnalysisTK.jar -T BaseRecalibrator -R f000_reference.fa -I f070_realigned_aligned.bam -knownSites /home/dmontaner/documentos/PRESENTACIONES/CursosBioinformatica/2013_09_cambridge/bundle/2.3/b37/dbsnp_137.b37.vcf -o 8_recalibration.table
-->


<!--
-knownSites bundle/hg18/dbsnp_132.hg18.vcf \
-knownSites another/optional/setOfSitesToMask.vcf \
-->



<h2 id="gatk-variant-discovery">GATK Variant Discovery</h2>
<p>SNP calling</p>
<pre><code>java -jar $gatk_path/GenomeAnalysisTK.jar -T UnifiedGenotyper -R f000_reference.fa -I f070_realigned_aligned.bam -glm SNP -o f080_snp_variants.vcf</code></pre>
<!--
-L bed file option
-->


<p>INDEL calling</p>
<pre><code>java -jar $gatk_path/GenomeAnalysisTK.jar -T UnifiedGenotyper -R f000_reference.fa -I f070_realigned_aligned.bam -glm INDEL -o f080_indel_variants.vcf</code></pre>
<!--
GATK Variant Data Analysis
--------------------------
-->



<h2 id="introduce-filters-in-the-vcf-file">Introduce filters in the VCF file</h2>
<p>Example: filter SNPs with low confidence calling (QD &lt; 12.0) and flag them as “LowConf”.</p>
<!--
QD:"Variant Confidence/Quality by Depth">
-->

<pre><code>java -jar $gatk_path/GenomeAnalysisTK.jar -T VariantFiltration -R f000_reference.fa -V f080_snp_variants.vcf --filterExpression &quot;QD &lt; 12.0&quot; --filterName &quot;LowConf&quot; -o f090_snp_filtered.vcf</code></pre>
</body>
</html>
