<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
  <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
  <meta http-equiv="Content-Style-Type" content="text/css" />
  <meta name="generator" content="pandoc" />
  <meta name="author" content="Quality control &amp; adapter trimming example" />
  <title>NGS data analisis course</title>
  <style type="text/css">code{white-space: pre;}</style>
  <link rel="stylesheet" href="../../../course_commons/css_template_for_examples.css" type="text/css" />
</head>
<body>
<div id="header">
<h1 class="title"><a href="http://ngscourse.github.io/">NGS data analisis course</a></h1>
<h2 class="author"><strong>Quality control &amp; adapter trimming example</strong></h2>
<h3 class="date"><em>(updated 05-11-2013)</em></h3>
</div>
<!-- COMMON LINKS -->

<h1 id="preliminaries">Preliminaries</h1>
<h2 id="software-used-in-this-practical">Software used in this practical</h2>
<ul>
<li><a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc">FastQC</a>: A quality control tool for high throughput sequence data.</li>
<li><a href="http://code.google.com/p/cutadapt">cutadapt</a>: A tool that removes adapter sequences from DNA sequencing reads.</li>
<li><a href="http://hannonlab.cshl.edu/fastx_toolkit">FASTX-Toolkit</a>: A collection of command line tools for Short-Reads FASTA/FASTQ files preprocessing.</li>
<li><a href="http://bowtie-bio.sourceforge.net/bowtie2/index.shtml">Bowtie 2</a>: Tool for aligning sequencing reads to long reference sequences.</li>
<li><a href="http://samtools.sourceforge.net">SAMtools</a>: Tool for aligning sequencing reads to long reference sequences.</li>
</ul>
<h2 id="data-used-in-this-practical">Data used in this practical</h2>
<ul>
<li><a href="../../../COURSE_EXAMPLE_DATA/mirbase_mature.fa">mirbase_mature.fa</a>: mature micro RNAs downloaded form mirbase</li>
</ul>
<p>You can download them or copy them to your <code>data</code> directory for the practical</p>
<!-- clean directory
    rm -r data
-->

<pre><code>mkdir data
cd data
cp ../../../../COURSE_EXAMPLE_DATA/f010_mirbase_mature.fa .</code></pre>
<p>  Find all data files for the course here: <a href="../../../COURSE_EXAMPLE_DATA">COURSE_EXAMPLE_DATA</a></p>
<h2 id="file-formats">File formats:</h2>
<p>FastQ:</p>
<ul>
<li><a href="http://en.wikipedia.org/wiki/FASTQ_format">Wikipedia</a>.</li>
<li><a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2847217">NAR 2010</a>.</li>
</ul>
<p>SAM/BAM</p>
<ul>
<li><a href="http://samtools.sourceforge.net/SAM1.pdf">SAMtools</a>.</li>
</ul>
<h1 id="exercise">Exercise</h1>
<p>The file <strong>f000_raw_mirna.fastq</strong> in the data folder contains reads form a microRNA sequencing experiment.</p>
<!--
    cd data
-->

<p>Use <code>wc</code> to count how many reads there are in the file (remember you have to divide by 4)</p>
<pre><code>wc -l f000_raw_mirna.fastq</code></pre>
<h2 id="raw-data-preprocessing">Raw Data Preprocessing</h2>
<p>Use fastqc to explore the quality of the data:</p>
<pre><code>fastqc f000_raw_mirna.fastq</code></pre>
<p>Find the <em>Overrepresented sequences</em> in <a href="http://www.mirbase.org/search.shtml">mirbase</a>.</p>
<p>There are 2 known adapters used in this experiment:</p>
<!--
    <<"COMMENT"
-->

<pre><code>CTGGGAAATCACCATAAACGTGAAATGTCTTTGGATTTGGGAATCTTATAAGTTCTGTATGAGACCACTCTAAAAA
CTTTTTTTCGTCCTTTCCACAAGATATATAAAGCCAAGAAATCGAAATACTTTCAAGTTACGGTAAGC</code></pre>
<!--
    COMMENT
-->



<p>Use R-Bioconductor to compute their reverse, complementary and reverse complementary.</p>
<p>Hint: <!--
    <<"COMMENT"
--></p>
<pre><code>library (Biostrings)
myseq &lt;- DNAString (&quot;CTTTTTTTCGTCCTTTCCACAAGATATATAAAGCCAAGAAATCGAAATACTTTCAAGTTACGGTAAGC&quot;)
reverse (myseq)
complement (myseq)
reverseComplement (myseq)</code></pre>
<!--
    cat (c (as.character (myseq),
            as.character (reverse (myseq)),
            as.character (complement (myseq)),
            as.character (reverseComplement (myseq))),
         sep = "\n")
-->

<!--
    COMMENT
-->



<p>Use grep form Linux shell to find out which of the versions of the adapter is in your data.</p>
<h2 id="adapter-1">Adapter 1</h2>
<pre><code>grep CTGGGAAATCACCATAAACGTGAAATGTCTTTGGATTTGGGAATCTTATAAGTTCTGTATGAGACCACTCTAAAAA f000_raw_mirna.fastq | wc -l  ## at the beginning is there
grep GACCCTTTAGTGGTATTTGCACTTTACAGAAACCTAAACCCTTAGAATATTCAAGACATACTCTGGTGAGATTTTT f000_raw_mirna.fastq | wc -l 
grep TTTTTAGAGTGGTCTCATACAGAACTTATAAGATTCCCAAATCCAAAGACATTTCACGTTTATGGTGATTTCCCAG f000_raw_mirna.fastq | wc -l  ## is there at the end ... but not so much
grep AAAAATCTCACCAGAGTATGTCTTGAATATTCTAAGGGTTTAGGTTTCTGTAAAGTGCAAATACCACTAAAGGGTC f000_raw_mirna.fastq | wc -l 

grep CTGGGAAATCACCATAAACGTGAAATGTCTTTGGA f000_raw_mirna.fastq | wc -l 
grep GACCCTTTAGTGGTATTTGCACTTTACAGAAACCT f000_raw_mirna.fastq | wc -l 
grep TTTTTAGAGTGGTCTCATACAGAACTTATAAGATT f000_raw_mirna.fastq | wc -l 
grep AAAAATCTCACCAGAGTATGTCTTGAATATTCTAA f000_raw_mirna.fastq | wc -l 

grep GAATCTTATAAGTTCTGTATGAGACCACTCTAAAAA f000_raw_mirna.fastq | wc -l 
grep CTTAGAATATTCAAGACATACTCTGGTGAGATTTTT f000_raw_mirna.fastq | wc -l 
grep ATCCAAAGACATTTCACGTTTATGGTGATTTCCCAG f000_raw_mirna.fastq | wc -l 
grep TAGGTTTCTGTAAAGTGCAAATACCACTAAAGGGTC f000_raw_mirna.fastq | wc -l </code></pre>
<h2 id="adapter-2">Adapter 2</h2>
<pre><code>grep CTTTTTTTCGTCCTTTCCACAAGATATATAAAGCCAAGAAATCGAAATACTTTCAAGTTACGGTAAGC f000_raw_mirna.fastq | wc -l   ## is there at the end ... but not so much
grep GAAAAAAAGCAGGAAAGGTGTTCTATATATTTCGGTTCTTTAGCTTTATGAAAGTTCAATGCCATTCG f000_raw_mirna.fastq | wc -l 
grep GCTTACCGTAACTTGAAAGTATTTCGATTTCTTGGCTTTATATATCTTGTGGAAAGGACGAAAAAAAG f000_raw_mirna.fastq | wc -l   ## is there at the beginning
grep CGAATGGCATTGAACTTTCATAAAGCTAAAGAACCGAAATATATAGAACACCTTTCCTGCTTTTTTTC f000_raw_mirna.fastq | wc -l 

grep CTTTTTTTCGTCCTTTCCACAAGATATATA f000_raw_mirna.fastq | wc -l 
grep GAAAAAAAGCAGGAAAGGTGTTCTATATAT f000_raw_mirna.fastq | wc -l 
grep GCTTACCGTAACTTGAAAGTATTTCGATTT f000_raw_mirna.fastq | wc -l 
grep CGAATGGCATTGAACTTTCATAAAGCTAAA f000_raw_mirna.fastq | wc -l 

grep AAGCCAAGAAATCGAAATACTTTCAAGTTACGGTAAGC f000_raw_mirna.fastq | wc -l 
grep TTCGGTTCTTTAGCTTTATGAAAGTTCAATGCCATTCG f000_raw_mirna.fastq | wc -l 
grep CTTGGCTTTATATATCTTGTGGAAAGGACGAAAAAAAG f000_raw_mirna.fastq | wc -l 
grep GAACCGAAATATATAGAACACCTTTCCTGCTTTTTTTC f000_raw_mirna.fastq | wc -l </code></pre>
<p>Use cutadapt to make an adapter trimming of the reads. Check the options:</p>
<ul>
<li><strong>-a</strong> for adapter to the 3’ end.</li>
<li><strong>-g</strong> for adapter to the 5’ end.</li>
</ul>
<p>command:</p>
<pre><code>cutadapt -g CTGGGAAATCACCATAAACGTGAAATGTCTTTGGATTTGGGAATCTTATAAGTTCTGTATGAGACCACTCTAAAAA -o f010_mirna_trim1.fastq f000_raw_mirna.fastq
cutadapt -a TTTTTAGAGTGGTCTCATACAGAACTTATAAGATTCCCAAATCCAAAGACATTTCACGTTTATGGTGATTTCCCAG -o f010_mirna_trim2.fastq f010_mirna_trim1.fastq
cutadapt -g GCTTACCGTAACTTGAAAGTATTTCGATTTCTTGGCTTTATATATCTTGTGGAAAGGACGAAAAAAAG         -o f010_mirna_trim3.fastq f010_mirna_trim2.fastq
cutadapt -a CTTTTTTTCGTCCTTTCCACAAGATATATAAAGCCAAGAAATCGAAATACTTTCAAGTTACGGTAAGC         -o f010_mirna_trim4.fastq f010_mirna_trim3.fastq</code></pre>
<h2 id="adapter-1-1">Adapter 1</h2>
<pre><code>grep CTGGGAAATCACCATAAACGTGAAATGTCTTTGGATTTGGGAATCTTATAAGTTCTGTATGAGACCACTCTAAAAA f010_mirna_trim4.fastq | wc -l  ## at the beginning is there
grep GACCCTTTAGTGGTATTTGCACTTTACAGAAACCTAAACCCTTAGAATATTCAAGACATACTCTGGTGAGATTTTT f010_mirna_trim4.fastq | wc -l 
grep TTTTTAGAGTGGTCTCATACAGAACTTATAAGATTCCCAAATCCAAAGACATTTCACGTTTATGGTGATTTCCCAG f010_mirna_trim4.fastq | wc -l  ## is there at the end ... but not so much
grep AAAAATCTCACCAGAGTATGTCTTGAATATTCTAAGGGTTTAGGTTTCTGTAAAGTGCAAATACCACTAAAGGGTC f010_mirna_trim4.fastq | wc -l 

grep CTGGGAAATCACCATAAACGTGAAATGTCTTTGGA f010_mirna_trim4.fastq | wc -l 
grep GACCCTTTAGTGGTATTTGCACTTTACAGAAACCT f010_mirna_trim4.fastq | wc -l 
grep TTTTTAGAGTGGTCTCATACAGAACTTATAAGATT f010_mirna_trim4.fastq | wc -l 
grep AAAAATCTCACCAGAGTATGTCTTGAATATTCTAA f010_mirna_trim4.fastq | wc -l 

grep GAATCTTATAAGTTCTGTATGAGACCACTCTAAAAA f010_mirna_trim4.fastq | wc -l 
grep CTTAGAATATTCAAGACATACTCTGGTGAGATTTTT f010_mirna_trim4.fastq | wc -l 
grep ATCCAAAGACATTTCACGTTTATGGTGATTTCCCAG f010_mirna_trim4.fastq | wc -l 
grep TAGGTTTCTGTAAAGTGCAAATACCACTAAAGGGTC f010_mirna_trim4.fastq | wc -l </code></pre>
<h2 id="adapter-2-1">Adapter 2</h2>
<pre><code>grep CTTTTTTTCGTCCTTTCCACAAGATATATAAAGCCAAGAAATCGAAATACTTTCAAGTTACGGTAAGC f010_mirna_trim4.fastq | wc -l   ## is there at the end ... but not so much
grep GAAAAAAAGCAGGAAAGGTGTTCTATATATTTCGGTTCTTTAGCTTTATGAAAGTTCAATGCCATTCG f010_mirna_trim4.fastq | wc -l 
grep GCTTACCGTAACTTGAAAGTATTTCGATTTCTTGGCTTTATATATCTTGTGGAAAGGACGAAAAAAAG f010_mirna_trim4.fastq | wc -l   ## is there at the beginning
grep CGAATGGCATTGAACTTTCATAAAGCTAAAGAACCGAAATATATAGAACACCTTTCCTGCTTTTTTTC f010_mirna_trim4.fastq | wc -l 

grep CTTTTTTTCGTCCTTTCCACAAGATATATA f010_mirna_trim4.fastq | wc -l 
grep GAAAAAAAGCAGGAAAGGTGTTCTATATAT f010_mirna_trim4.fastq | wc -l 
grep GCTTACCGTAACTTGAAAGTATTTCGATTT f010_mirna_trim4.fastq | wc -l 
grep CGAATGGCATTGAACTTTCATAAAGCTAAA f010_mirna_trim4.fastq | wc -l 

grep AAGCCAAGAAATCGAAATACTTTCAAGTTACGGTAAGC f010_mirna_trim4.fastq | wc -l 
grep TTCGGTTCTTTAGCTTTATGAAAGTTCAATGCCATTCG f010_mirna_trim4.fastq | wc -l 
grep CTTGGCTTTATATATCTTGTGGAAAGGACGAAAAAAAG f010_mirna_trim4.fastq | wc -l 
grep GAACCGAAATATATAGAACACCTTTCCTGCTTTTTTTC f010_mirna_trim4.fastq | wc -l </code></pre>
<p>Check the quality again using fastqc:</p>
<pre><code>fastqc f010_mirna_trim4.fastq</code></pre>
<p>Use cutadapt to filter reads by quality and length. Check the options:</p>
<ul>
<li><strong>-q</strong> quality cutoff.</li>
<li><strong>-m</strong> minimum length.</li>
<li><p><strong>-M</strong> maximum length.</p>
<p>cutadapt -m 17 -M 30 -q 10 -o f020_mirna_cut.fastq f010_mirna_trim4.fastq</p></li>
</ul>
<p>And check quality again using fastqc</p>
<pre><code>fastqc f020_mirna_cut.fastq</code></pre>
<p>Find again the <em>Overrepresented sequences</em> in <a href="http://www.mirbase.org/search.shtml">mirbase</a>.</p>
<p>Count how many reads are left for the analysis (divide by 4)</p>
<pre><code>wc -l f000_raw_mirna.fastq
wc -l f020_mirna_cut.fastq</code></pre>
<h2 id="make-a-non-redundant-dataset">Make a non-redundant dataset</h2>
<p>Use fastx to to create a non-redundant fasta file with the reads by collapsing identical sequences.</p>
<pre><code>fastx_collapser -Q33 -v -i f020_mirna_cut.fastq -o f030_mirna_collapsed.fasta</code></pre>
<p>The parameter <code>-Q33</code> indicates that we are using Illumina encoded quality scores, not Sanger encoding.</p>
<p>See the first lines of the new file <code>030_mirna_collapsed.fasta</code></p>
<ul>
<li>¿Can you find the number of times each read is replicated?</li>
<li>Find the most common sequence. Find it in the results form the FastQC analysis.</li>
</ul>
<h2 id="format-mirbase">Format miRBase</h2>
<p>Download the <strong>mature</strong> microRNA sequences form the <a href="http://www.mirbase.org/ftp.shtml">miRBase web page</a>.</p>
<p>Filter the human microRNAs. Generally you will write an script for this kind of task: Python, Perl, R-Bioconductor… but in this very easy case you can directly use Linux shell command <code>grep</code>.</p>
<pre><code>grep -A 1 &quot;Homo sapiens&quot; mature.fa &gt; mature_human0.fa
grep -v  &quot;^--&quot; mature_human0.fa &gt; mature_human1.fa </code></pre>
<p>where <code>-A</code> indicates how many lines should be returned after the line matching the pattern. <code>-A</code> introduces some separator lines (<code>--</code>) that are removed in the <code>grep -v</code> line.</p>
<p>Observe that the database we downloaded form miRBase is a database of <strong>RNA</strong> we will need to convert it into a <strong>DNA</strong> fasta file. You can use fastx for this purpose.</p>
<pre><code>fasta_nucleotide_changer -i mature_human1.fa -o mature_human.fa -d</code></pre>
<h2 id="map-against-mirbase-using-bowtie2">Map against miRBase using bowtie2</h2>
<p>Fist we need to <strong>build an index</strong> for bowtie:</p>
<pre><code>bowtie2-build mature_human.fa mature_human</code></pre>
<p>And now we can run the <strong>alignment step</strong>:</p>
<pre><code>bowtie2-align -x mature_human -f f030_mirna_collapsed.fasta -S f040_mirna_mapped.sam</code></pre>
<!-- bowtie2 AND bowtie2-align ARE THE SAME -->


<p>Explore the <strong>SAM</strong> file that has been created. <a href="http://samtools.sourceforge.net/SAM1.pdf">Here</a> you can find the SAM format specifications. More information is available in <a href="http://samtools.sourceforge.net">SAMtools home page</a>.</p>
<h2 id="use-samtools-to-explore-the-alignment">Use SAMtools to explore the alignment</h2>
<p>The file <code>f040_mirna_mapped.sam</code> is a text file. You can open it with a text editor or visualize it in the shell using commands such as <code>less</code>, <code>head</code> or <code>tail</code>.</p>
<p>The SAM (Sequence Alignment/Map) format is described <a href="http://samtools.sourceforge.net/SAM1.pdf">here</a>. It is basically a TAB-delimited file with an optional <strong>header section</strong> and an <strong>alignment section</strong>.</p>
<p>The alignment of each read is described in a row of the file <strong>11 fields</strong>:</p>
<ul>
<li>QNAME: read name</li>
<li>FLAG: bitwise flag (See <a href="http://picard.sourceforge.net/explain-flags.html">here</a> for interpretation)</li>
<li>RNAME: chromosome</li>
<li>POS: leftmost genomic position</li>
<li>MAPQ: mapping quality (<a href="http://en.wikipedia.org/wiki/Phred_quality_score">Phred scale</a>)</li>
<li>CIGAR: CIGAR string (gaps, clipping)</li>
<li>RNEXT: paired read name</li>
<li>PNEXT: paired read position</li>
<li>TLEN: total length of template</li>
<li>SEQ: read base sequence</li>
<li>QUAL: read base quality</li>
</ul>
<p>We can use <a href="http://samtools.sourceforge.net/samtools.shtml">SAMtools</a> to explore and analyze the results of our mapping. Read the <a href="http://samtools.sourceforge.net/samtools.shtml">SAMtools manual</a> for details on the available tools.</p>
<p>You can use samtools to extract the header of the SAM file:</p>
<pre><code>samtools view -SH f040_mirna_mapped.sam </code></pre>
<p>Or the alignment section without the header:</p>
<pre><code>emacs f040_mirna_mapped.sam
samtools view -S f040_mirna_mapped.sam &gt; f041_mirna_mapped_nohead.sam
emacs f041_mirna_mapped_nohead.sam</code></pre>
<p>Or the number of alignment rows of the SAM file:</p>
<pre><code>wc -l f041_mirna_mapped_nohead.sam</code></pre>
<p>You could alternatively use <code>grep</code> instead of SAMtools to count the number of alignments:</p>
<pre><code>grep -v &quot;^@&quot; f041_mirna_mapped_nohead.sam | wc -l</code></pre>
<hr />
<p>Remember the fasta file with the reads we aligned :</p>
<pre><code>head -n 2 f030_mirna_collapsed.fasta </code></pre>
<p>And find the read names in the sam file:</p>
<pre><code>head -n 1 f041_mirna_mapped_nohead.sam</code></pre>
<hr />
<p>The SAM file is a text format. It may be transformed into a binary format using SAMtools. The resulting file is called BAM file. The binary format is desirable in order to speed up computations and efficiency. It is required by some of the tools in SAMtools and in many third-party software.</p>
<p>Transform to BAM format:</p>
<pre><code>samtools view -bS -o f050_mirna_mapped.bam f040_mirna_mapped.sam</code></pre>
<p>you can now still access to all the information in the file as before</p>
<pre><code>samtools view -H f050_mirna_mapped.bam            ## the header
samtools view    f050_mirna_mapped.bam  | head    ## the alignment</code></pre>
<p>but now you will not be able to read it with text editors, or shell commands as <code>less</code>, <code>head</code> or <code>tail</code>.</p>
<hr />
<p>The BAM files can be prepared for efficient computation by:</p>
<ul>
<li><p><strong>Sorting</strong> the alignments in them</p>
<pre><code>samtools sort f050_mirna_mapped.bam f060_mirna_mapped_sorted   ## no extension in the output prefix</code></pre></li>
<li><p><strong>Indexing</strong> the file.</p>
<pre><code>samtools index f060_mirna_mapped_sorted.bam     ## creates a .bai file     </code></pre></li>
</ul>
<hr />
<p>We can now get some statistics about our mapping</p>
<pre><code>samtools idxstats f060_mirna_mapped_sorted.bam &gt; f070_mirna_mapped_stats.txt</code></pre>
<p><code>idxstats</code> produces a TAB delimited file each line consisting of</p>
<ul>
<li>reference sequence name</li>
<li>sequence length</li>
<li>number of mapped reads</li>
<li>unmapped reads</li>
</ul>
<p>You can check that the number of reads assigned to the to the first sequence is correct:</p>
<pre><code>head f070_mirna_mapped_stats.txt
grep hsa-let-7a-5p f041_mirna_mapped_nohead.sam | wc -l</code></pre>
<!--

We now have a sorted BAM file called ec_snp.sorted.bam. 
Sorted BAM is a useful format because the alignments are both compressed, 
which is convenient for long-term storage, 
and sorted, which is convenient for variant discovery. 
Finally, we call variants from the Sorted BAM:

#    samtools mpileup -f f030_mirna_collapsed.fasta f060_mirna_mapped_sorted.bam > f070_mirna_pileup.bcf
-->



<h1 id="homework">Homework</h1>
<ul>
<li><p>Increase the parameter <code>-N</code> in bowtie and see whether more reads align to the reference of miRNAs.<br /><!--
```
    bowtie2-align -x mature_human -N 1 -f f030_mirna_collapsed.fasta -S f040_mirna_mapped_N1.sam
```
--></p></li>
<li><p>Align the uncollapsed fastq file against the human microRNAs.<br /><!--
```
    bowtie2 -x mature_human -U f020_mirna_cut.fastq -S f045_mirna_cut_mapped.sam
```
--></p></li>
<li><p>Align the collapsed and uncollapsed files against all miRNAs in miRBase.<br /><!--
```
    ##prepare the reference
    fasta_nucleotide_changer -i mature.fa -o matureDNA.fa -d
    bowtie2-build matureDNA.fa matureDNA

    ##align collapsed
    bowtie2-align -x matureDNA -f f030_mirna_collapsed.fasta -S g040_mirna_mapped.sam

    ##align uncollapsed
    bowtie2-align -x matureDNA -U f020_mirna_cut.fastq -S g045_mirna_cut_mapped.sam
```
--></p></li>
</ul>
<!-- ---------------------------------------------------------------------------
References
================================================================================
-->
</body>
</html>
