<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
  <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
  <meta http-equiv="Content-Style-Type" content="text/css" />
  <meta name="generator" content="pandoc" />
  <meta name="author" content="Title of the practical" />
  <title>NGS data analysis course</title>
  <style type="text/css">code{white-space: pre;}</style>
  <link rel="stylesheet" href="../../../course_commons/css_template_for_examples.css" type="text/css" />
</head>
<body>
<div id="header">
<h1 class="title"><a href="http://ngscourse.github.io/">NGS data analysis course</a></h1>
<h2 class="author"><strong>Title of the practical</strong></h2>
<h3 class="date"><em>(updated 23-02-2014)</em></h3>
</div>
<!-- COMMON LINKS HERE -->

<h1 id="preliminaries">Preliminaries</h1>
<h2 id="software-used-in-this-practical">Software used in this practical:</h2>
<ul>
<li><a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/" title="notes on hover">FastQC</a> : (SPACE BEFORE THE COLON) Explain …</li>
<li><a href="http://bio-bwa.sourceforge.net/" title="notes BWA">BWA software</a> : Explain with a sentence.</li>
</ul>
<h2 id="file-formats-explored">File formats explored:</h2>
<p>FastQ:</p>
<ul>
<li><a href="http://en.wikipedia.org/wiki/FASTQ_format">Wikipedia</a></li>
<li>more</li>
</ul>
<p>SAM/BAM</p>
<ul>
<li>links</li>
</ul>
<h2 id="data-used-in-this-practical">Data used in this practical</h2>
<ul>
<li><a href="../../../COURSE_EXAMPLE_DATA/f010_mirbase_mature.fa">mirbase_mature.fa</a>: mature micro RNAs downloaded form mirbase</li>
</ul>
<p>You can download them or copy them to your <code>data</code> directory for the practical</p>
<!-- clean directory
    rm -r data
-->

<pre><code>mkdir data
cd data
cp ../../../../COURSE_EXAMPLE_DATA/f010_mirbase_mature.fa .</code></pre>
<p> </p>
<p>Find all data files for the course here: <a href="../../../COURSE_EXAMPLE_DATA">COURSE_EXAMPLE_DATA</a></p>
<h1 id="exercise">Exercise</h1>
<p>The file <strong>f000_raw_mirna.fastq</strong> in the data folder contains reads form a microRNA sequencing experiment.</p>
<p>Use <code>wc</code> to count how many reads there are in the file (remember you have to divide by 4)</p>
<pre><code>wc -l f000_raw_mirna.fastq</code></pre>
<h2 id="raw-data-preprocessing">Raw Data Preprocessing</h2>
<p>Use fastqc to explore the quality of the data:</p>
<pre><code>fastqc f000_raw_mirna.fastq</code></pre>
<p>Find the <em>Overrepresented sequences</em> in [mirbase][mirbase-search].</p>
<p>There are 2 known adapters used in this experiment:</p>
<pre><code>CTGGGAAATCACCATAAACGTGAAATGTCTTTGGATTTGGGAATCTTATAAGTTCTGTATGAGACCACTCTAAAAA
CTTTTTTTCGTCCTTTCCACAAGATATATAAAGCCAAGAAATCGAAATACTTTCAAGTTACGGTAAGC</code></pre>
<p>Use R-Bioconductor to compute their reverse, complementary and reverse complementary.</p>
<pre><code>library (Biostrings)
myseq &lt;- DNAString (&quot;CTTTTTTTCGTCCTTTCCACAAGATATATAAAGCCAAGAAATCGAAATACTTTCAAGTTACGGTAAGC&quot;)
reverse (myseq)
complement (myseq)
reverseComplement (myseq)</code></pre>
<h1 id="further-work">Further work</h1>
<p>Some ideas</p>
</body>
</html>
